### PR TITLE
Move wait_for_dataset_collection to populators

### PR DESCRIPTION
**Pull request by mvdbeek**
This will allow us us to reuse this in other tests I'm working on (enhancing job_search to work with collection in/out).

### CODE REVIEW

File: test/api/test_dataset_collections.py [hunk 1]
```diff
@@ -113,11 +111,10 @@ def test_list_download(self):
 
     def test_pair_download(self):
         dataset_collection = self.dataset_collection_populator.create_pair_in_history(self.history_id).json()
+        self.dataset_collection_populator.wait_for_dataset_collection(dataset_collection, assert_ok=True)
         returned_dce = dataset_collection["elements"]
         assert len(returned_dce) == 2, dataset_collection
         returned_datasets = [dce["object"] for dce in returned_dce]
```
Comment (nsoranzo) [inline test/api/test_dataset_collections.py]: test/api/test_dataset_collections.py:117:9: F841 local variable 'returned_datasets' is assigned to but never used

Comment (mvdbeek) [reply test/api/test_dataset_collections.py]: Thanks @nsoranzo, I rebased with that change.

File: test/api/test_dataset_collections.py [hunk 2]
```diff
@@ -97,11 +97,9 @@ def test_create_list_of_new_pairs(self):
 
     def test_list_download(self):
         dataset_collection = self.dataset_collection_populator.create_list_in_history(self.history_id).json()
+        self.dataset_collection_populator.wait_for_dataset_collection(dataset_collection, assert_ok=True)
         returned_dce = dataset_collection["elements"]
-        assert len(returned_dce) == 3, dataset_collection
-        returned_datasets = [dce["object"] for dce in returned_dce]
-        for element in returned_datasets:
-            self.dataset_populator.wait_for_dataset(history_id=self.history_id, dataset_id=element['id'], assert_ok=True)
+        assert len(dataset_collection["elements"]) == 3, dataset_collection
```
Comment (nsoranzo) [inline test/api/test_dataset_collections.py]: You can probably revert the change to this line back to:
```python
assert len(returned_dce) == 3, dataset_collection
```
for consistency with the other tests.

Comment (mvdbeek) [reply test/api/test_dataset_collections.py]: Right, will do!

**Review by nsoranzo — APPROVED** [review]
